staging/src/k8s.io/client-go/tools/portforward/portforward.go

```diff
@@ -299,17 +300,27 @@ func (pf *PortForwarder) getListener(protocol string, hostname string, port *For
 // waitForConnection waits for new connections to listener and handles them in
 // the background.
 func (pf *PortForwarder) waitForConnection(listener net.Listener, port ForwardedPort) {
+	var tempDelay time.Duration // how long to sleep on accept failure
 	for {
 		select {
 		case <-pf.streamConn.CloseChan():
 			return
 		default:
 			conn, err := listener.Accept()
 			if err != nil {
-				// TODO consider using something like https://github.com/hydrogen18/stoppableListener?
-				if !strings.Contains(strings.ToLower(err.Error()), "use of closed network connection") {
-					runtime.HandleError(fmt.Errorf("error accepting connection on port %d: %v", port.Local, err))
+				if ne, ok := err.(net.Error); ok && ne.Temporary() {
```

Comment:
https://github.com/golang/go/issues/45729

> It sounds like deprecate means "leave things alone, comment that it's not to be used, and stop discussing it".

:upside_down_face:

Reply:
After our discussion above, I think that there is a consensus that "retry mechanism needs to be used". Although golang currently sets Temporary to Deprecated (the reason is because Temporary() is not well defined), it does not use ne.Temporary () We can't confirm whether we should retry after an error. So I think the code of this PR is appropriate. When the Golang community releases a mechanism to explicitly detect Temporary errors, the code can be improved.

Reply:
oh, I think just the contrary https://github.com/kubernetes/kubernetes/pull/111743#issuecomment-1209147053 ,`Temporary()` is going to be deprecated and not recommended to be used.
Opinions on the golang mailing list have the same thinking as I do,  retrying on fds limits errors is not safe

Reply:
I think `kubectl port-forward` is a special proxy mechanism [access to Pod's web service through kube-apiserver]. And the proxy itself should continue to try to service subsequent connections after running out of files. Because the service provider is not a `port-forward` but a process inside the Pod. The service may encounter a short access peak. During the peak period, the newly established connection cannot provide services due to the limitation of FD, but the service should resume normal operation after the peak.

ps: Similar mechanism we can see in traefik, nginx. We can test this result by

````
/tmp » docker run --ulimit nofile=20:20 -it nginx:1.23.1
...
2022/08/09 14:16:46 [crit] 32#32: accept4() failed (24: Too many open files)
2022/08/09 14:16:47 [crit] 32#32: accept4() failed (24: Too many open files)
2022/08/09 14:16:47 [crit] 33#33: accept4() failed (24: Too many open files)
2022/08/09 14:16:47 [crit] 32#32: accept4() failed (24: Too many open files)
2022/08/09 14:16:48 [crit] 33#33: accept4() failed (24: Too many open files)
2022/08/09 14:16:48 [crit] 32#32: accept4() failed (24: Too many open files)
2022/08/09 14:16:48 [crit] 33#33: accept4() failed (24: Too many open files)
2022/08/09 14:16:48 [crit] 32#32: accept4() failed (24: Too many open files)
2022/08/09 14:16:49 [crit] 33#33: accept4() failed (24: Too many open files)
2022/08/09 14:16:49 [crit] 32#32: accept4() failed (24: Too many open files)
2022/08/09 14:16:49 [crit] 33#33: accept4() failed (24: Too many open files)
2022/08/09 14:16:49 [crit] 32#32: accept4() failed (24: Too many open files)
2022/08/09 14:16:50 [crit] 33#33: accept4() failed (24: Too many open files)
2022/08/09 14:16:50 [crit] 32#32: accept4() failed (24: Too many open files)
2022/08/09 14:16:50 [crit] 33#33: accept4() failed (24: Too many open files)
2022/08/09 14:16:50 [crit] 32#32: accept4() failed (24: Too many open files)
2022/08/09 14:16:55 [crit] 32#32: *8 open() "/usr/share/nginx/html/index.html" failed (24: Too many open files), client: 172.17.0.1, server: localhost, request: "GET / HTTP/1.1", host: "172.17.0.2"
2022/08/09 14:16:55 [crit] 32#32: *8 open() "/usr/share/nginx/html/50x.html" failed (24: Too many open files), client: 172.17.0.1, server: localhost, request: "GET / HTTP/1.1", host: "172.17.0.2"
172.17.0.1 - - [09/Aug/2022:14:16:55 +0000] "GET / HTTP/1.1" 500 177 "-" "curl/7.58.0" "-"
2022/08/09 14:16:56 [crit] 32#32: *9 open() "/usr/share/nginx/html/index.html" failed (24: Too many open files), client: 172.17.0.1, server: localhost, request: "GET / HTTP/1.1", host: "172.17.0.2"
2022/08/09 14:16:56 [crit] 32#32: *9 open() "/usr/share/nginx/html/50x.html" failed (24: Too many open files), client: 172.17.0.1, server: localhost, request: "GET / HTTP/1.1", host: "172.17.0.2"
172.17.0.1 - - [09/Aug/2022:14:16:56 +0000] "GET / HTTP/1.1" 500 177 "-" "curl/7.58.0" "-"
2022/08/09 14:16:57 [crit] 32#32: *10 open() "/usr/share/nginx/html/index.html" failed (24: Too many open files), client: 172.17.0.1, server: localhost, request: "GET / HTTP/1.1", host: "172.17.0.2"
2022/08/09 14:16:57 [crit] 32#32: *10 open() "/usr/share/nginx/html/50x.html" failed (24: Too many open files), client: 172.17.0.1, server: localhost, request: "GET / HTTP/1.1", host: "172.17.0.2"
172.17.0.1 - - [09/Aug/2022:14:16:57 +0000] "GET / HTTP/1.1" 500 177 "-" "curl/7.58.0" "-"
2022/08/09 14:16:58 [crit] 32#32: *11 open() "/usr/share/nginx/html/index.html" failed (24: Too many open files), client: 172.17.0.1, server: localhost, request: "GET / HTTP/1.1", host: "172.17.0.2"
2022/08/09 14:16:58 [crit] 32#32: *11 open() "/usr/share/nginx/html/50x.html" failed (24: Too many open files), client: 172.17.0.1, server: localhost, request: "GET / HTTP/1.1", host: "172.17.0.2"
172.17.0.1 - - [09/Aug/2022:14:16:58 +0000] "GET / HTTP/1.1" 500 177 "-" "curl/7.58.0" "-"
2022/08/09 14:16:59 [crit] 32#32: *12 open() "/usr/share/nginx/html/index.html" failed (24: Too many open files), client: 172.17.0.1, server: localhost, request: "GET / HTTP/1.1", host: "172.17.0.2"
2022/08/09 14:16:59 [crit] 32#32: *12 open() "/usr/share/nginx/html/50x.html" failed (24: Too many open files), client: 172.17.0.1, server: localhost, request: "GET / HTTP/1.1", host: "172.17.0.2"
172.17.0.1 - - [09/Aug/2022:14:16:59 +0000] "GET / HTTP/1.1" 500 177 "-" "curl/7.58.0" "-"
```

As we have seen, nginx will continue to serve after the client connection is released.

Reply:
The current implementation here is also unreasonable. port-forward exits the `accept-loop` after reaching the file descriptor limit. Because the listener is not closed, port-forward is still listening on the port. But at this point port-forward can no longer work, at least we should exit the process after giving a prompt.

Reply:
I think that we already greatly discussed, documented and expressed our opinions with links and references.
I defer to current code maintainers to decide

Reply:
OK, thank you for your suggestions and relevant materials, and thank you again for your work on this PR.